### DISCUSSION TITLE
Missing Files When Building U-Boot for Creating a Bootable MMC Device

### DISCUSSION THREAD

**Discussion by b-sullender** [inindev/odroid-m1#11] [Q&A]
I'm trying to build u-boot and create a bootable mmc device, but looking at your code for creating the Debian image leaves me with some questions.

Where did you get the files
`https://github.com/inindev/odroid-m1/releases/download/v12.0.1/idbloader.img`
and
`https://github.com/inindev/odroid-m1/releases/download/v12.0.1/u-boot.itb`?

When i build u-boot following the instructions at `https://wiki.odroid.com/odroid-m1/software/building_u-boot` i don't see those files.

These are the files i have after building u-boot
```
bl31_0x00040000.bin
bl31_0x00069000.bin
bl31_0x0006b000.bin
bl31_0xfdcc1000.bin
bl31_0xfdcce000.bin
bl31_0xfdcd0000.bin
bl31.elf
config.mk
idblock.bin
Kbuild
Kconfig
MAINTAINERS
Makefile
make.sh
PREUPLOAD.cfg
README
rk356x_spl_loader_v1.18.112.bin
snapshot.commit
System.map
tee.bin
u-boot
u-boot.bin
u-boot.cfg
u-boot.cfg.configs
u-boot.dtb
u-boot-dtb.bin
uboot.img
u-boot.lds
u-boot.map
u-boot-nodtb.bin
u-boot.srec
u-boot.sym
usb_update.txt
```

**Comment by inindev** [comment]
Are you building mainline u-boot?

Here is the script to build it:
https://github.com/inindev/odroid-m1/blob/main/uboot/make_uboot.sh

```sh make_uboot.sh```

The script is intended to be run from an arm64 linux device.

**Reply by b-sullender** [reply]
I'm building U-Boot from `https://github.com/hardkernel/u-boot/tree/odroidm1-v2017.09` on my x86-64 PC using the instructions I mentioned earlier.

Using your script doesn't quite fit my needs, as I want to create additional partitions on my device. Specifically, I plan to have `/var/www` on a separate partition and possibly implement a mirror RAID using an SSD.

I'm also doing this to learn the entire process.

However, I'm unsure where to obtain the `idbloader.img` and `u-boot.itb` files.

**Comment by inindev** [comment]
I am not going to be of much help in building the proprietary 2017 downstream u-boot from Rockchip.  You have to also run Rockchip x86 binaries as well.  The output targets from Rockchip look to be ```u-boot.itb``` and possibly ```u-boot-spl.bin```.

The whole purpose of my repos is to build unmodified up-stream distributions directly from the source u-boot and debian repositories.

This should help: https://opensource.rock-chips.com/wiki_U-Boot

**Reply by b-sullender** [reply]
> The whole purpose of my repos is to build unmodified up-stream distributions directly from the source u-boot and debian repositories.

I overlooked that your script was using `https://github.com/u-boot/u-boot` and not `https://github.com/hardkernel/u-boot`. My mistake.

> The output targets from Rockchip look to be `u-boot.itb` and possibly `u-boot-spl.bin`

So building from `https://github.com/u-boot/u-boot` should produce `idbloader.img` and `u-boot.itb`. Right? (different names)

> You have to also run Rockchip x86 binaries as well

I attempted to compile `https://github.com/u-boot/u-boot` on ODROID-M1 without including `BL31="$atf_file" ROCKCHIP_TPL="$tpl_file"` and as you would expect i got errors:

`Image 'simple-bin' is missing external blobs and is non-functional: rockchip-tpl atf-bl31`.

Did you download these files or compile them. If you compiled them, then what repository's did you use?

**Comment by b-sullender** [comment]
The rockchip files needed for compiling u-boot for the M1 can be downloaded from the rkbin repository `https://github.com/rockchip-linux/rkbin/`. It will have the latest updates. Search the repository for `rk3568_bl31` and `rk3568_ddr_1560MHz` to find the files.

**Comment by inindev** [comment]
I have updated the rkbin binaries from ```https://github.com/rockchip-linux/rkbin``` and have new builds you may be interested in:

**debian bookworm v12.5 arm64 - rc4**
https://github.com/inindev/debian-image/releases

**u-boot arm64 inindev v2024.07-rc4**
https://github.com/inindev/uboot-rockchip/releases
